core/src/test/scala/org/apache/spark/api/python/PythonRDDSuite.scala

```diff
@@ -76,12 +79,22 @@ class PythonRDDSuite extends SparkFunSuite with LocalSparkContext {
   }
 
   test("python server error handling") {
-    val authHelper = new SocketAuthHelper(new SparkConf())
-    val errorServer = new ExceptionPythonServer(authHelper)
-    val client = new Socket(InetAddress.getLoopbackAddress(), errorServer.port)
-    authHelper.authToServer(client)
-    val ex = intercept[Exception] { errorServer.getResult(Duration(1, "second")) }
-    assert(ex.getCause().getMessage().contains("exception within handleConnection"))
+    val savedSparkEnv = SparkEnv.get
+    try {
+      val conf = new SparkConf()
+      val env = mock(classOf[SparkEnv])
+      doReturn(conf).when(env).conf
+      SparkEnv.set(env)
```

Comment:
@gaborgsomogyi, sorry for an ignorant question. How does this test the current patch? BTW, I personally think it's fine with manual tests.

Reply:
It's not testing the newly added code in any way. The test blowed-up with NPE because `SparkEnv.get` introduced in the server side code.

Reply:
I've considered to add automated test for this but measuring time and making assertions based on that is always scary to me. It could be the next flaky test which could make upset the guys.

Reply:
@gaborgsomogyi . If that case, shall we make a separate JIRA and PR for that?
> It's not testing the newly added code in any way.

Reply:
Does this exist in `branch-3.0/2.4`?
> The test blowed-up with NPE because SparkEnv.get introduced in the server side code.